report_broken_clients.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+    "fmt"
+    "os/exec"
+    "strings"
+    "io/ioutil"
+    "net/http"
+    "net/url"
+
+    "github.com/groob/mackit/cfpref"
+)
+
+func serialNumber() string {
+    out, err := exec.Command("/usr/sbin/ioreg", "-l").Output()
+    if err != nil {
+        return ""
+    }
+    for _, l := range strings.Split(string(out), "\n") {
+        if strings.Contains(l, "IOPlatformSerialNumber") {
+            s := strings.Split(l, " ")
+            t := strings.Replace(s[len(s)-1], "\"", "", -1)
+            return t
+        }
+    }
+    return ""
+}
+
+
+func main() {
+    serial_number:= serialNumber()
+    if serial_number == "" {
+        panic("Could not retrieve serial number")
+    }
+    sal_url := cfpref.CopyAppValue("ServerURL", "com.github.salopensource.sal")
+    key := cfpref.CopyAppValue("key", "com.github.salopensource.sal")
+
+    //Testing
+    // sal_url := "http://127.0.0.1:8000"
+    // Now you all know the key for my test group on my computer.
+    // key := "2k4wohv0sopmzkfp5ocubal47iwds8ddz8h4w7kjbcdjmbyhlgiztdyj7e8up048l89u35vqt08l4jrrkmaegj0j9ba264g3r80geybog0e9e2wrik6jqow7wh4fb5az"
+
+    the_url := fmt.Sprintf("%v/checkin/", sal_url)
```

Comment:
instead of using sprintf here a more reliable way of working with urls is to call `url.Parse` on sal_url and then add  url.Path to the returned object https://golang.org/pkg/net/url/#Parse

Reply:
Like

```
theUrl := url.Parse(salUrl)
theUrl.Path = "/checkin/"
```

?